src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/security/MaintainUserCheck.kt

```diff
@@ -18,35 +23,41 @@ class MaintainUserCheck(
   }
 
   @Throws(GroupRelationshipException::class)
-  fun ensureMaintainerGroupRelationship(
+  suspend fun ensureMaintainerGroupRelationship(
     userName: String?,
     groupCode: String,
   ) {
     // if they have maintain privileges then all good
-    if (AuthenticationFacade.hasRoles("ROLE_MAINTAIN_OAUTH_USERS")) {
+    if (authenticationFacade.hasRoles("ROLE_MAINTAIN_OAUTH_USERS")) {
       return
     }
-    val maintainer =
-      userRepository.findByUsername(userName).orElseThrow()
+    val maintainer = userService.getUserAndGroupByUserName(userName)
     // otherwise group managers must have a group in common for maintenance
-    if (maintainer.groups.none { it.groupCode == groupCode }) {
-      // no group in common, so disallow
-      throw GroupRelationshipException(groupCode, "Group not with your groups")
+    if (maintainer != null) {
+      if (maintainer.groups.none { it.groupCode == groupCode }) {
+        // no group in common, so disallow
+        throw GroupRelationshipException(groupCode, "Group not with your groups")
+      }
     }
   }
 
   @Throws(UserGroupRelationshipException::class)
-  fun ensureUserLoggedInUserRelationship(loggedInUser: String?, authorities: Collection<GrantedAuthority>, user: User) {
+  suspend fun ensureUserLoggedInUserRelationship(loggedInUser: String?, authorities: Collection<GrantedAuthority>, user: User): User? {
     // if they have maintain privileges then all good
     if (canMaintainUsers(authorities)) {
-      return
+      return null
     }
     // otherwise group managers must have a group in common for maintenance
-    val loggedInUserEmail = userRepository.findByUsername(loggedInUser).orElseThrow()
-    if (Sets.intersection(loggedInUserEmail.groups, user.groups).isEmpty()) {
+    val loggedInUserEmail = userService.getUser(loggedInUser)
+    Optional.of(loggedInUserEmail!!).orElseThrow()
```

Comment:
Do we need the check here as userService.getUser checks and throws the same exception ?

Reply:
TODO - investigate and tidy up use of orElseThrow()